packages/frontend/app/routes/webhooks.tsx

```diff
@@ -0,0 +1,164 @@
+    if (selectedTypeSet.size > 0) {
+      searchParams.set('type', Array.from(selectedTypeSet).join(','))
+    } else {
+      searchParams.delete('type')
+    }
+    setSearchParams(searchParams)
+  }
+
+  return (
+    <>
+      <div className='pt-4 flex flex-col space-y-8'>
+        <div className='flex flex-col rounded-md bg-offwhite px-6'>
+          <PageHeader>
+            <div className='flex-1'>
+              <h3 className='text-2xl leading-10'>Webhook Events</h3>
+            </div>
+          </PageHeader>
+          <div className='p-1'>
+            <h3 className='text-lg font-bold'>Filters</h3>
+            <div className='flex items-center'>
+              <PopoverFilter
+                label='Event type'
+                values={type ? type : ['all']}
+                options={[
+                  {
+                    name: 'All',
+                    value: 'all',
+                    action: () => {
+                      navigate(``)
+                    }
+                  },
+                  ...Object.values(WebhookEventType).map((value) => ({
+                    name:
+                      value.charAt(0).toUpperCase() +
+                      value.slice(1).replace(/[_.]/g, ' '),
+                    value: value,
+                    action: () => {
+                      setTypeFilterParams(value)
+                    }
+                  }))
+                ]}
+              />
+            </div>
+          </div>
+          <Table>
+            <Table.Head columns={['ID', 'Type', 'Data']} />
+            <Table.Body>
+              {webhooks.edges.length ? (
+                webhooks.edges.map((webhook) => (
+                  <Table.Row key={webhook.node.id}>
+                    <Table.Cell>{webhook.node.id}</Table.Cell>
+                    <Table.Cell>{webhook.node.type}</Table.Cell>
+                    <Table.Cell>
+                      <Button
+                        aria-label='view webhook data'
+                        state={{ data: webhook.node.data }}
+                        to={'/webhooks/data'}
+                      >
+                        View data
+                      </Button>
+                    </Table.Cell>
+                  </Table.Row>
+                ))
+              ) : (
+                <Table.Row>
+                  <Table.Cell colSpan={4} className='text-center'>
+                    No webhook events found.
+                  </Table.Cell>
+                </Table.Row>
+              )}
+            </Table.Body>
+          </Table>
+          <div className='flex items-center justify-between p-5'>
+            <Button
+              aria-label='go to previous page'
+              disabled={!webhooks.pageInfo.hasPreviousPage}
+              onClick={() => {
+                searchParams.delete('after')
+                searchParams.set('before', previousPageUrl)
+                setSearchParams(searchParams)
+              }}
```

Comment:
If we construct `previousPageUrl` on the server, we can directly call `navigate(previousPageUrl)`.

Reply:
From what I read my understanding is that it's better to use setSearchParams when you're only changing the search parameters. What is the advantage of using navigate here instead?